clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/ApplicationEventProcessor.java

```diff
@@ -113,16 +113,15 @@ public void process(ApplicationEvent event) {
                 process((UnsubscribeApplicationEvent) event);
                 return;
 
+            case CONSUMER_REBALANCE_LISTENER_CALLBACK_COMPLETED:
```

Comment:
The logic in `ApplicationEventProcessor` is _supposed_ to be glue code between the event from the application thread and resources in the background thread. As such, the core logic for that interaction with the background thread resource is _supposed_ to live in the test for the resource.

That said, it couldn't hurt to have some tests here. It would likely surface any regressions faster than trying to piece together which end of the glued bits isn't acting correctly.

Is this something for which I can file a Jira and we can tackle in a separate PR?

Reply:
Sure. We can tackle it in a separate PR.